### PR TITLE
Feature/proxmox

### PR DESCRIPTION
**Pull request by bemanuel**
Creating Packer to use with Proxmox VE

### CODE REVIEW

File: package/packer/proxmox/cloud/config/cloud.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+ssh_pwauth: yes
```
Comment (dweomer) [inline package/packer/proxmox/cloud/config/cloud.yml:1]: `ssh_pwauth` is not supported by k3OS.

File: package/packer/proxmox/cloud/.gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+stages:
```
Comment (dweomer) [inline package/packer/proxmox/cloud/.gitlab-ci.yml:1]: Please remove this file `.gitlab-ci.yml`. It is fine to carry it as a patch in a fork but we don't use gitlab-ci and do not plan to.

File: package/packer/proxmox/cloud/README.md [hunk 3]
```diff
@@ -0,0 +1 @@
+# Template k3os to Proxmox VE
```
Comment (dweomer) [inline package/packer/proxmox/cloud/README.md:1]: Please provide a sample build command in the readme, e.g.
```bash
packer build -var-file=variables.json k3os-cloud.json
```

See https://github.com/rancher/k3os/blob/master/package/packer/vagrant/README.md as an example.

File: package/packer/proxmox/cloud/config/cloud.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      LoginGraceTime          30s
+      MaxAuthTries            5
+
+      Ciphers chacha20-poly1305@openssh.com,aes256-gcm@openssh.com,aes128-gcm@openssh.com,aes256-ctr,aes192-ctr,aes128-ctr
+      MACs hmac-sha2-512-etm@openssh.com,hmac-sha2-256-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-512,hmac-sha2-256,umac-128@openssh.com
+      KexAlgorithms curve25519-sha256@libssh.org,diffie-hellman-group-exchange-sha256
+      Subsystem   sftp    internal-sftp
+
+k3os:
+  password: YourSSHPassword
```
Comment (dweomer) [inline package/packer/proxmox/cloud/config/cloud.yml:20]: The `password` line seems unnecessary.

**Review by dweomer — COMMENTED** [review]
@bemanuel thank you for contributing this patch. I've left a few comments. If you could address these I will be able to accept the change request.

**Comment by dweomer** [comment]
Additionally, I don't understand the sub-directory hierarchy. Is this necessary to make the proxmox builder happy?

**Comment by bemanuel** [comment]
> Additionally, I don't understand the sub-directory hierarchy. Is this necessary to make the proxmox builder happy?

The initial idea was create 3 different templates (cloud, agent, server), but cloud-init isn't working. I refactor the hierarchy, now it's ok.

File: package/packer/proxmox/cloud/config/cloud.yml (hunk 4 above)
Comment (bemanuel) [reply package/packer/proxmox/cloud/config/cloud.yml]: For some reason it's necessary.

File: package/packer/proxmox/cloud/README.md (hunk 3 above)
Comment (bemanuel) [reply package/packer/proxmox/cloud/README.md]: Done.

File: package/packer/proxmox/cloud/config/cloud.yml (hunk 1 above)
Comment (bemanuel) [reply package/packer/proxmox/cloud/config/cloud.yml]: Ah, ok. Sorry. Corrected.

**Pull request merged**